iree/compiler/Codegen/LLVMCPU/test/vector_contract_to_arm_asm.mlir

```diff
@@ -297,3 +298,25 @@ func @mmt_1x4x8_i8i8i32_vecmat(
 // AARCH64-DOTPROD-SAME:      {{((.*sdot){2})}}
 // AARCH64-DOTPROD-SAME:      "{{(\=w,){2}(w,){3}0,1}}"
 // AARCH64-DOTPROD-SAME:      {{\((vector<16xi8>, ){2}(vector<4xi8>, ){1}(vector<4xi32>(, )?){2}\)}}
```

Comment:
I like this idea so I started typing it out, and that made me realize one downside: it has _quadratic complexity_. Namely:  the number of testcases here is proportional to the number of CPU feature variants that we have kernels for. So if we make each testcase test all the runs, of which we have also one per CPU feature variant, it becomes quadratic.  This could be mitigated like you say by merely checking that there is a `vector.contract` so it is quadratic but "with a small multiplicative constant". Even that could get cumbersome, e.g. imagine the future PR adding the 8th CPU variant, having to add stub testcases like this for the testcases of the existing 7 variants.

I could be convinced the other way if one can spell out some kind of bug that could slip in as long as we don't extend the tests in this way...